telegram/tests/test_telegram.py

```diff
@@ -0,0 +1,344 @@
+import json
+
+from twisted.internet.defer import inlineCallbacks, returnValue, DeferredQueue
+from twisted.web.server import NOT_DONE_YET
+from twisted.web import http
+
+from vumi.tests.utils import LogCatcher
+from vumi.tests.helpers import VumiTestCase
+from vumi.tests.fake_connection import FakeHttpServer
+from vumi.transports.httprpc.tests.helpers import HttpRpcTransportHelper
+
+from telegram.telegram import TelegramTransport
+
+
+class TestTelegramTransport(VumiTestCase):
+
+    @inlineCallbacks
+    def setUp(self):
+        self.helper = self.add_helper(
+            HttpRpcTransportHelper(TelegramTransport)
+        )
+        self.request_queue = DeferredQueue()
+        self.pending_requests = []
+        self.addCleanup(self.finish_requests)
+        self.mock_server = FakeHttpServer(self.handle_inbound_request)
+
+        self.default_user = {
+            'id': 'default_user_id',
+            'username': '@default_user',
+        }
+        self.bad_telegram_response = {
+            'ok': False,
+            'description': 'Bad request',
+        }
+        self.bot_username = '@bot'
+        self.API_URL = 'https://api.telegram.org/bot'
+        self.TOKEN = '1234'
```

Comment:
Things like `self.API_URL` that have constant values can be class attributes (like `CONFIG_CLASS` or `transport_name` in the transport class) instead of being set in the `setUp()` method -- `setUp()` is more for things that need to be created fresh in each test.